packages/core/src/Cardano/util/phase2Validation.ts

```diff
@@ -0,0 +1,4 @@
+import { OnChainTx } from '../types';
+
+export const isPhase2ValidationErrTx = ({ inputSource }: Pick<OnChainTx, 'inputSource'>): boolean =>
```

Comment:
Updated here: [f60e3b8](https://github.com/input-output-hk/cardano-js-sdk/pull/607/commits/f60e3b827f24657725d2f6b1f8f15c1589552482)

Reply:
Might be a comment mismatch, as that commit doesn't address the feedback, but when reviewing again your name sounds better than my suggestion. As minor feedback I'd expect that function to take the whole tx object, otherwise you may as well just check the property inline. How about this:

``` ts
export const txFailedPhase2Validation = (tx: OnChainTx): boolean =>
```

Reply:
I prefer requesting as input only what is actually needed. Otherwise, we always require a full OnChainTx object to call it when we only need `inputSource`. Maybe we'll want to call it for a reduced tx object, and we wouldn't be able to do that.
The reason why I went with the `Pick` version was to allow adding more properties easily in the future and also to hint to the object normally having this property.

Reply:
I like the `Pick` variant better too. In addition to what @mirceahasegan said, this style also makes tests cleaner: no need to specify entire object or explicitly cast partial objects into the full ones.

Reply:
The utility makes sense to me as a way to abstract all of the internals, so it's use case **_is_** when you have the full tx to pass in :smile:

I'm fine with either of these option here, so just treat this as a discussion around style.

Reply:
Yes, you will be passing full tx, and the style with `Pick` allows that. The only downside I can see with `Pick` is readability of the function signature.

Reply:
I reverted the commit that changes the name: [165c369](https://github.com/input-output-hk/cardano-js-sdk/pull/607/commits/165c36965e6ad5a7da09eeb0762aefeec2ff3395)

Reply:
I usually do `git commit --fixup amend:[hash] --no-verify` for message updates (--no-verify is needed because commit message linter doesn't allow amend fixups)